### ISSUE TITLE
About the use of soft_label

### ISSUE THREAD

**Issue by BaiMeiyingxue** [PingCheng-Wei/DepressionEstimation#9]
Hi PingCheng-Wei,
    I was wondering why to use the function torch.log() to convert the **probs**, but the article claims _predict the probability between the 4 different depressive classes of the corresponding subscore with the softmax-function:_ seems that softmax function is used instead.

the location of code is in https://github.com/PingCheng-Wei/DepressionEstimation/blob/c8c402fb1d3052cbe2fb5fba674b61685a167734/models/Audio_ConvLSTM/utils.py#LL417C1-L417C1

Looking forward to your reply~

**Comment by BaiMeiyingxue** [comment]
Hi PingCheng-Wei,
Have you ever occurred a situation where the model performed well in the training stage, in which the TPR and TNR are 0.79 and 0.94 respectively, but in the test stage TPR is close to zero, and the TNR is close to 1, the model predicts all samples in test dataset to negative, and the predicted score is always a fixed number in per clips at the participant level(soft_label/sam are used). I'm confused and could you give me some advice?

**Comment by PingCheng-Wei** [comment]
Hi BaiMeiyingxue, It seems like you are facing either an overfitting problem in terms of training data or the lack of convergence of the model, meaning that even you have trained the model, the model still performs pretty weird and like random guessing. Since the test stage result is like that, I think it’s more likely you are facing the second one. Indeed, I have also faced this problem at that time while I was working on this. What this result basically means is that your model is not really well-trained enough or in other words, your trained model is too bias, toward non-depressed class. In fact, 70% or even more of the participants in the “training dataset” are not depressed at all and only less than 30% of participant have major depression. “Test dataset”, on the other hand, has relative equal amount of participants from both classes if I have the right memory. This indicates that if you don’t try to come up a way to solve the bias problem in training dataset, you will always have a very bias model that will perform really worst on the test dataset. In fact, you could have a model that also only predict training data to negative (high TNR) but yield a high TPR and TNR because 70% of data in the training dataset is actually negative (accuracy 70%). However, this doesn’t mean your model is good or well-trained and you will see the bad result  later on test data, which I believe is your case. You can try to print out the result from training dataset and see whether the model also only predict negative as a result.  That’s why I applied those two techniques, MUSDL and SAM, to cope with this problem, which you can see in the paper. I recommend you also applied these two methods in your model as well as training process. Of course, if you have other better idea, feel free to explore more. But basically what you have to try to solve right now is the bias in the dataset. Other ideas that could be useful are weighted cross entropy loss or “WeightedRandomSampler” from pytorch to load the data (batch) from each class equally. I have applied all 4 aforementioned techniques to finally solve this problem. Hope this information help!  All the best with your research!  Best, Ping-Cheng Wei     Sent from Mail for Windows From: BaiMeiyingxueSent: 2023年5月17日 下午 05:55To: PingCheng-Wei/DepressionEstimationCc: SubscribedSubject: Re: [PingCheng-Wei/DepressionEstimation] About the use of soft_label (Issue #9) Hi PingCheng-Wei,Have you ever occurred a situation where the model performed well in the training stage, in which the TPR and TNR are 0.79 and 0.94 respectively, but in the test stage TPR is close to zero, and the TNR is close to 1, the model predicts all samples in test dataset to negative, and the predicted score is fixed per clips in participant level. I'm confused and could you give me some advice?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***> 乾淨無病毒。www.avg.com

**Comment by BaiMeiyingxue** [comment]
Hi PingCheng-Wei,
Thanks for your patient explanation, your suggestion make great sense to me!
